torch/_inductor/constant_folding.py

```diff
@@ -210,12 +210,30 @@ def constant_fold(gm, constraint_fn: Optional[Callable[[torch.fx.Node], bool]] =
         replace_node_with_constant(gm, node, constant)
 
     erased_params = []
+    # Get all attr users by looking up the graph instead from node.users, because in this case
+    # _tensor_constant0 and _tensor_constant0_1 are actually refereing to the same tensor.
+
+    #     opcode         name                 target            args                         kwargs
+    # -------------  -------------------  ----------------  ---------------------------  --------
+    # placeholder    arg0_1               arg0              ()                           {}
+    # get_attr       _tensor_constant0    state             ()                           {}
+    # call_function  add                  aten.add.Tensor   (arg0_1, _tensor_constant0)  {}
+    # get_attr       _tensor_constant0_1  state             ()                           {}
+    # call_function  add_                 aten.add_.Tensor  (_tensor_constant0_1, 1)     {}
+    # output         output               output            ([add],)                     {}
```

Comment:
Where does this graph come from and why is it not functionalized?

Reply:
The graph is from `capture_pre_autograd_graph`. It's not functionalized because of the in place mutation ops. In the future, after `capture_pre_autograd_graph` is migrated to torch.export(..., predispath=True)`, I guess it can be functionalized by using the `copy` node. It's just not happening yet and we need to fix the edge case here.

Reply:
fyi Inductor's fx passes generally assume that the graph has already been functionalized, with the possible exception of `copy_` nodes which are retained in inference graphs.

Reply:
how can it be functionalized with the `get_attr` node which may hold data?